Kotlin/Maths/HappyNumber.kt

```diff
@@ -0,0 +1,79 @@
+/*
+kotlin program to check happy numbers
+
+A happy number is a number, if we reach to 1 after replacing the number by the squares of the digits of the number
+
+Following examples will help to understand better:-
+
+Input: n = 19
+Output: True
+19 is Happy Number,
+1^2 + 9^2 = 82
+8^2 + 2^2 = 68
+6^2 + 8^2 = 100
+1^2 + 0^2 + 0^2 = 1
+As we reached to 1, 19 is a Happy Number.
+
+Input: n = 20
+Output: False
+20 is not a happy number
+2^2 + 0^2 = 4
+4^2 = 16
+As we cannot reach to 1, 20 is not a happy number
```

Comment:
Remove this 
- Add a beginner-friendly description at the top

Reply:
I updated the snippet as per the instructions. Please check it again.